### DISCUSSION TITLE
how to show user name with relationship?

### DISCUSSION THREAD

**Discussion by jererosas** [Power-Components/livewire-powergrid#1403] [Q&A]
I need to show the user name

Before I did it this way: `<td>$registers->relationship->name</td> `

In the documentation it says to do it with "->with('users')"

 public function datasource(): Builder
`     return Reservas::query()
            ->where('dia_id', Session::get('evento'))->where('tipo', 'online')->with('users');`

Now how can I do it with powergrid?

**Comment by dansysanalyst** [comment]
HI @jererosas 

If I understood well, you should be able to do something like this:

Data source:

```php 
    public function datasource(): Builder
    {
        return Reservas::query()->with('users'); //skipping conditions for brevity
    }
```

Data source fields:

```php
    public function fields(): PowerGridFields
    {
        return PowerGrid::fields()
            ->add('id')
            ->add('codigo')
            ->add('user_id')
            ->add('reservante', fn(Reservas $reserva) => $reserva->user->name)
            ->add('created_at');
    }
```

Columns:

```php
    public function columns(): array
    {
        return [
            Column::make('Id', 'id'),
            Column::make('Codigo Reserva', 'codigo'),
            Column::make('Reservante', 'reservante', 'user_id'),
        ];
    }
```

**Reply by ZulfikarHD** [reply]
i seems can't get it to work using `fields()` and `Column::make('title','field_name','db_column_name')` it's always return nothing, so i decided using `addColumn()` and it's work. did i miss something ?
